reference/local-development.md

```diff
@@ -0,0 +1,128 @@
+# Local development
+
+A Turso database [instance] runs an open-source server called [sqld]. This
+server:
+
+- Handles incoming client connections over HTTP and websockets
+- Offers a JSON-based wire protocol for clients
+- Connects the client to the embedded [libSQL] database engine (a fork of
+  SQLite) to execute queries
+- Manages the underlying database data in a [SQLite database
+  file](#about-sqlite-database-files)
+
+Turso adds additional features and configurations on top of sqld:
+
+- DNS and secure sockets with SSL
+- Client authentication with database tokens provided by the [Turso CLI]
+- World-wide instance location support
+- Serverless instance scaling
+- Automatic replication between instances
+
+For daily development, you might be satisfied with the core features provided by
+sqld, without the additional features offered by Turso. You might also want to
+avoid any costs incurred by working with sqld instances managed by Turso. For
+these cases, you have three options for building and testing client code written
+with the [libSQL client SDKs]:
+
+- [Read and write local SQLite data files directly](#use-local-sqlite-database-files)
+- [Run sqld locally using the Turso CLI](#run-sqld-using-the-turso-cli)
+- [Build and run an instance of sqld locally](#build-and-run-sqld-separately)
+
+## Use local SQLite database files
```

Comment:
I find it useful to add some wording about which kind of environments this can be used with.
You do mention the Javascript client restriction, but that's already a consequence, not a cause.

The real divide is environments with or without a filesystem, although we may find a better way to phrase it, potentially even mentioning some by name (like cloudflare workers).

There's also a case to be made that the SQLite database files should come first, since it is the simplest, although I see the point of the current order, since at the moment most of our incoming traffic is from cloudflare.

Reply:
@glommer SQLite database files does come first of the three options. Or am I misunderstanding what you mean here?

Reply:
It comes first compared to sqld, but there is a large explanation about what sqld even is before the file scenario.

For the users to whom the file version is preferred, do they even need to know about sqld ?

Reply:
They need to know that Turso/sqld uses SQLite database files, which is part of the explanation, even if a small part of it.

I added a bit about environments with filesystems in a new paragraph.  If that's OK, I'm incline to go ahead and publish this now. Then, when the new Turso capabilities page is available, move the introduction content to that page and link to it from here.

Reply:
it's ok from my side to publish